File: app/src/main/java/ru/nikshlykov/englishwordsapp/SortWordsDialogFragment.java
package ru.nikshlykov.englishwordsapp;

import android.app.Dialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class SortWordsDialogFragment extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        String[] array = {"По алфавиту", "По сложности"};
        return new AlertDialog.Builder(getActivity())
                .setTitle(R.string.dialog___sort_words___title)
                .setSingleChoiceItems(array, 1, null)
                .setPositiveButton(R.string.yes, null)
                .setNegativeButton(R.string.cancel, null)
                .create();
        // ДОБАВИТЬ
        // РАБОТУ
        // С ВАРИАНТАМИ СОРТИРОВКИ
        // !!!!!!!!!!!!!!!
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return super.onCreateView(inflater, container, savedInstanceState);
    }
}

File: feature-groups/src/main/java/ru/nikshlykov/feature_groups_and_words/ui/fragments/SortWordsDialogFragment.kt
package ru.nikshlykov.feature_groups_and_words.ui.fragments

import android.app.Dialog
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import ru.nikshlykov.feature_groups_and_words.R
import ru.nikshlykov.core_ui.R as CoreUiR

internal class SortWordsDialogFragment : DialogFragment() {
  private var sortParam = 0

  interface SortWordsListener {
    fun sort(sortParam: Int)
  }

  private var sortWordsListener: SortWordsListener? = null

  fun setSortWordsListener(sortWordsListener: SortWordsListener) {
    this.sortWordsListener = sortWordsListener
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    sortParam = arguments?.getInt(EXTRA_SORT_PARAM) ?: BY_ALPHABET
    Log.d("SortWordsDialogFragment", "sortParam from bundle = $sortParam")
  }

  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
    val sortParams = resources
      .getStringArray(R.array.preference_entries___sort_words_in_subgroup)
    return AlertDialog.Builder(requireContext())
      .setTitle(R.string.dialog___sort_words___title)
      .setSingleChoiceItems(sortParams, sortParam) { dialog, which -> sortParam = which }
      .setPositiveButton(CoreUiR.string.yes) { _, _ -> sortWordsListener?.sort(sortParam) }
      .setNegativeButton(CoreUiR.string.cancel, null)
      .create()
  }

  companion object {
    // TODO refactoring. Подобные места можно поменять на enum.
    // Возможные значения параметра сортировки.
    const val BY_ALPHABET = 0
    const val BY_PROGRESS = 1

    const val EXTRA_SORT_PARAM = "SortParam"
  }
}